django/db/migrations/autodetector.py

```diff
@@ -1051,6 +1053,7 @@ def arrange_for_graph(self, changes, graph, migration_name=None):
                     migration.dependencies.append(app_leaf)
                 if i == 0 and not app_leaf:
                     new_name = "0001_%s" % migration_name if migration_name else "0001_initial"
+                    migration.initial = True
```

Comment:
Hmm, I may not be fully understanding all this code, but am I understanding right that this is technically redundant, because the app should not be in the previous state at all, so it should have already had `initial` set to `True` in `_detect_changes()`? Or are there cases where that would not be the case?